### PR TITLE
drops are randomly selected, not based on user values fixes #7929

### LINKED ISSUE #7929: Change Drop Calculation To Not Rely on Seeded Stats from User
General Info
- UUID: c1d236c4-cc77-443a-9830-265af88bd7fb
- Browser: Chome
- OS: Windows 7
### Description

I have a grey wishing well task, and I was using it. I got a lion cub egg about 16 times in a row (I overused since I was curious about the behavior :P). I then paused, and tried a different task. I then got recurring shade potions...without a pause, I went to the first task I was using. Still shade potions...didn't seem to be any sync errors, and I wasn't fixing my gold value in between clicks to keep it the same (which Alys tells me would give me the same drop).

### PR DESCRIPTION
**Pull request by gambtho**
Fixes https://github.com/HabitRPG/habitrpg/issues/7929
### Changes

Uses math.random instead of predictable random from drop calculations

Previous discussion - https://github.com/HabitRPG/habitrpg/pull/7967#issuecomment-246126330

---

UUID:  349ea7be-f378-4593-8702-672f75fa83f9

### CODE REVIEW

File: common/script/fns/randomVal.js [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 
 // Get a random property from an object
 // returns random property (the value)
 
-module.exports = function randomVal (user, obj, options = {}) {
+function randomGenerator (user, seed, providedRandom) {
+  return providedRandom ? providedRandom(user, seed) : Math.random();
+}
+
+module.exports = function randomVal (user, obj, options = {}, randomFunc) {
```
Comment (crookedneighbor) [inline common/script/fns/randomVal.js]: randomFunc should be in the options object, not a fourth argument.

File: test/common/fns/randomVal.test.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import randomVal from '../../../common/script/fns/randomVal';
```
Comment (crookedneighbor) [inline test/common/fns/randomVal.test.js]: It's correct to rename this file to `randomVal.test.js`, but to make reviewing the changes easier, let's leave it as is.

**Comment by gambtho** [comment]
any other changes needed here?

**Comment by crookedneighbor** [comment]
There's one thing I want to check first. Stand by.

**Comment by crookedneighbor** [comment]
I thought this issue might be blocking this https://github.com/HabitRPG/habitrpg/issues/7851, but after re-reading it, I don't think it is.

**Comment by gambtho** [comment]
fixed merge conflict

**Comment by crookedneighbor** [comment]
Thanks for your work on this. I've noted this toward your next tier.

**Pull request closed without merging**